commit 7e6404e11e06eae79b246c16ee270435e607b9d5 (refs/changes/08/2008/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-05-26T11:00:23-07:00 (2 years, 4 months ago)
    
    joyent/node-smartdc-auth#10 bump the signaturecache event emitter limit

diff --git a/lib/keyring.js b/lib/keyring.js
index 6316ef5..8b734ee 100644
--- a/lib/keyring.js
+++ b/lib/keyring.js
@@ -205,6 +205,11 @@ function SignatureCache(opts) {
 
 	this.expiry = opts.expiry || 10000;
 	this.pending = new EventEmitter();
+	/*
+	 * Turn off warnings about EventEmitter leaks here; we might have lots
+	 * of things waiting for the same signature at once.
+	 */
+	this.pending.setMaxListeners(0);
 	this.pending.table = {};
 	this.table = {};
 	this.list = [];
diff --git a/package.json b/package.json
index cb2e44f..6b00179 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "smartdc-auth",
     "description": "Joyent Authentication Library",
-    "version": "2.5.4",
+    "version": "2.5.5",
     "author": "Joyent (joyent.com)",
     "contributors": [
         "Mark Cavage",
